pkg/cloudprovider/provider/openstack/provider.go

```diff
@@ -148,6 +148,25 @@ func (p *provider) getConfig(s v1alpha1.ProviderSpec) (*Config, *providerconfigt
 		klog.V(6).Infof("Region from configuration or environment variable not found")
 	}
 
+	// Load timeout
+	instanceReadyCheckPeriod, err := p.configVarResolver.GetConfigVarIntValueOrEnv(rawConfig.InstanceReadyCheckPeriod, "INSTANCE_READY_CHECK_PERIOD")
```

Comment:
Instead of plain int type and introducing new family of functions (GetConfigVarInt...), I'd prefer for https://golang.org/pkg/time/#ParseDuration to be used on a string value.

Reply:
Ah, very good catch! I was unhappy with the int too, as I felt it was too ambiguous in regarding what it represents (timeout in seconds or minutes).